File: java/src/main/java/io/fivexer/sdk/model/NotificationSequence.java
package io.fivexer.sdk.model;

import java.util.List;

/**
 * A stored notification sequence.
 *
 * <p>Populated by Gson via field reflection and exposed read-only; callers receive instances
 * from the client and never construct them.
 */
public class NotificationSequence {
    private String id;
    private String name;
    private boolean enabled;
    private List<NotificationSequenceStep> steps;
    private List<String> filterTags;
    private String createdAt;  // ISO-8601
    private String updatedAt;  // ISO-8601

    public String getId() { return id; }
    public String getName() { return name; }
    public boolean isEnabled() { return enabled; }
    public List<NotificationSequenceStep> getSteps() { return steps; }
    public List<String> getFilterTags() { return filterTags; }
    public String getCreatedAt() { return createdAt; }
    public String getUpdatedAt() { return updatedAt; }
}

File: php/src/Model/NotificationSequence.php
<?php

declare(strict_types=1);

namespace Fivexer\SDK\Model;

use Fivexer\SDK\Internal\Json;

/**
 * A stored notification sequence.
 *
 * Response model: built from a decoded JSON array and read-only thereafter.
 */
final class NotificationSequence
{
    public function __construct(
        public readonly string $id,
        public readonly string $name,
        public readonly bool $enabled,
        /** @var list<NotificationSequenceStep> */
        public readonly array $steps,
        /** @var list<string> */
        public readonly array $filterTags,
        /** ISO-8601 */
        public readonly string $createdAt,
        /** ISO-8601 */
        public readonly string $updatedAt,
    ) {
    }

    /**
     * @param array<string, mixed> $data
     * @internal
     */
    public static function fromArray(array $data): self
    {
        return new self(
            id: (string) ($data['id'] ?? ''),
            name: (string) ($data['name'] ?? ''),
            enabled: (bool) ($data['enabled'] ?? false),
            steps: Json::parseEach($data, 'steps', [NotificationSequenceStep::class, 'fromArray']),
            filterTags: \array_map('strval', $data['filterTags'] ?? []),
            createdAt: (string) ($data['createdAt'] ?? ''),
            updatedAt: (string) ($data['updatedAt'] ?? ''),
        );
    }
}
